packages/schemas/src/schemaComponents.ts

```diff
@@ -57,12 +57,12 @@ export const confidenceIntervalConfigSchema = Type.Object({
 	help: Type.Optional(Type.String()),
 });
 
-export const componentConfigSchemas: Record<InputComponent, TObject> = {
+export const componentConfigSchemas = {
 	[InputComponent.checkbox]: checkboxConfigSchema,
 	[InputComponent.textArea]: textAreaConfigSchema,
 	[InputComponent.textInput]: textInputConfigSchema,
 	[InputComponent.datePicker]: datePickerConfigSchema,
 	[InputComponent.fileUpload]: fileUploadConfigSchema,
 	[InputComponent.memberSelect]: memberSelectConfigSchema,
 	[InputComponent.confidenceInterval]: confidenceIntervalConfigSchema,
-} as const;
+} as const satisfies Record<InputComponent, TObject>;
```

Comment:
This is the change that actually allows the type inference to work, typescript now recognizes the unique type of each key, rather than them all being `TObject`